org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/groovy/SmartApp.groovy

```diff
@@ -202,26 +214,35 @@ def unregisterDeviceChange() {
 		httpError(500, "something went wrong: $e")
 	}
 
-	log.info "Current subscription map is ${state.deviceSubscriptionMap}" 
+	log.info "Current subscription map is ${state.deviceSubscriptionMap}"
+    log.info "Current verification key map is ${state.verificationKeyMap}"
 }
 
 //Endpoints function: Subscribe to device additiona/removal updated in a location
 def registerDeviceGraph() {
 	def subscriptionEndpt = params.subscriptionURL
-
+    
     if (subscriptionEndpt != null && subscriptionEndpt != "undefined"){
         subscribe(location, "DeviceCreated", locationEventHandler, [filterEvents: false])
         subscribe(location, "DeviceUpdated", locationEventHandler, [filterEvents: false])
         subscribe(location, "DeviceDeleted", locationEventHandler, [filterEvents: false])
 
     	if(state.locationSubscriptionMap[location.id] == null){
             state.locationSubscriptionMap.put(location.id, [subscriptionEndpt])
-            log.info "Added subscription URL: ${subscriptionEndpt} for Location ${location.name}"
         }else if (!state.locationSubscriptionMap[location.id].contains(subscriptionEndpt)){
             state.locationSubscriptionMap[location.id] << subscriptionEndpt
-            log.info "Added subscription URL: ${subscriptionEndpt} for Location ${location.name}"
         }
+        
+        if(params.key != null){
+            if(state.verificationKeyMap[subscriptionEndpt] == null){
+                state.verificationKeyMap.put(subscriptionEndpt, params.key)
```

Comment:
The "params.key != null" is. I realized there is some code duplication, so the updated version should fix it.
The backwards compatibility part is also at line 295 and 328.

---
In reply to: [112324653](https://github.com/openT2T/translators/pull/234#discussion_r112324653) [](ancestors = 112324653)